gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/db/FabricDatabase.java

```diff
@@ -1252,20 +1261,86 @@ else if (qVal instanceof AbstractGfxdReplayableMessage) {
         }
       }
 
-      for (GemFireContainer container : uninitializedContainers) {
-        if (logger.infoEnabled() &&
-            !Misc.isSnappyHiveMetaTable(container.getSchemaName())) {
-          logger.info("FabricDatabase: start initializing container: "
-              + container);
+      // In case of reconnect, same cache is used and many locks are
+      // already taken by reconnect thread.
+      // Can't do initlization in different thread.
+      if (Thread.currentThread().getName().equals("ReconnectThread")) {
```

Comment:
Ok. This new code block is to handle initialization in parallel?
Can you please add a bigger comment above and explain briefly the need and then the logic of doing it.

Factoring the entire in a different private method will be cleaner.

Reply:
Yes, one thread at a time. Will refactor.
One concern was whether to limit the threadpool size?

Reply:
Yes. May be 50 is good enough. Too small will defeat the purpose.